### DISCUSSION TITLE
Search performance, sometimes slow to load or even needs a page refresh

### DISCUSSION THREAD

**Discussion by SinghGurd** [microsoft-search/pnp-modern-search#2261] [Q&A]
Is there anyway to improve the search performance? We are using the version 4.6.1.0 with 'Local SharePoint People Results' using a custom layout template and have the graph toolkit and people hover card enabled. Sometimes search is really slow and even appears to be stuck which gets resolved by doing a page refresh.

Also, we are using a custom layout template stored in a Sharepoint Library. Would it make any difference if we simply modify the code for custom layout rather than using a template URL.

When using the browser developer extension and testing in a network throttling mode of 'Slow 3G', I have noticed it takes a long time from the moment query is entered in search box to when the loading spinner starts in search results. Not sure what is causing that delay as I assumed the search results web part should start showing the loading spinner as soon as the query is entered in search box.

Steps to re-produce are below with a search box and its connected search results web part doing a people search with custom layout template stored in SharePoint library.

1. Use the browser developer tools to throttle the network speed to a slow connection
2. Type the search term and press Enter
3. HTTP 304 appears for layout template under networking tab of dev tools. However, nothing happens in search results web part, it stays as-is displaying the previous results at least more than 15 seconds
5. The search result loading spinner starts and postquery HTTP call can be seen under networking tab of developer tools
6. Results starts to load.

**Comment by FranckyC** [comment]
This can have many causes. Here is what you can try see if it improves:
- Use a inline template instead of an external URL
- Dont use the existing `<pnp-persona>` component in your template if present, prefer a custom component.

**Reply by SinghGurd** [reply]
Thanks @FranckyC...I tracked it down to search verticals web part. Essentially, all the behaviour noted in my original post occurs when the page have a search verticals webpart. Once the verticals web part is removed, the search result refresh as soon as enter is pressed in search box.

**Reply by wobba** [reply]
@FranckyC do you think the web part connections can cause this with dynamic data?

**Reply by FranckyC** [reply]
@SinghGurd could you share your Web Part settings ,especially:

- Tab values for verticals
- Search results query template value
- Connection tab in the search results

It may be a miss configuration in the connection resulting to some endless loop or something like it. The verticals WP is really a dummy Web Part so it should not slow anything.
